src/commands/common/issue-update.ts

```diff
@@ -0,0 +1,220 @@
+ *    `([\d\D]*)(?=<!-- probot:Status -->)` We set this Anchor to catch message that we want to replace. This RegExp means `([\d\D]*)` is followed by `<!-- probot:Status -->`.
+ *    `[\r\n]*[\d\D]` match any blank character and followed by any character
+ *
+ *
+ * Example:
+ * ```js
+ * const STATUS_REGEX = /(Status).*[\r\n]+([\d\D]*)(?=<!-- probot:Status -->)[\r\n]*[\d\D]*\/i;
+ * const issueBody = `## Status
+ * this is an example.<!-- probot:Status -->
+ * other characters which is not important.
+ * `;
+ * const StatusData = issueBody.match(STATUS_REGEX); //new notification
+ * console.log(StatusData);
+ * console.log(StatusData[2]);
+ * ```
+ * Output:
+ * ```
+ * [
+ *  'Status\n' +
+ *  'this is an example.<!-- probot:Status -->\n' +
+ *  'other characters which is not important.\n',
+ *  'Status',
+ *  'this is an example.',
+ *  index: 3,
+ *  input: '## Status\n' +
+ *  'this is an example.<!-- probot:Status -->\n' +
+ *  'other characters which is not important.\n',
+ *  groups: undefined
+ * ]
+ * this is an example.
+ * ```
+ *
+ */
+
+const STATUS_REGEX = /(Status).*[\r\n]+([\d\D]*)(?=<!-- probot:Status -->)[\r\n]*[\d\D]*/i;
+
+/**
+ * Find the Notification title text in the file. And catch the text which is before the `<!-- probot:Notification -->` (this is an comment, so it is invisible)
+ *
+ * Catch the text which is before the `<!-- probot:Notification -->`
+ *
+ *    `[\r\n]` match any blank characters
+ *    `([\d\D]*)` match any character, which will be replaced that we expect.
+ *    `([\d\D]*)(?=<!-- probot:Notification -->)` We set this Anchor to catch message that we want to replace. This RegExp means `([\d\D]*)` is followed by `<!-- probot:Notification -->`.
+ *    `[\r\n]*[\d\D]` match any blank character and followed by any character
+ *
+ * Example:
+ * ```js
+ * const STATUS_REGEX = /(:warning:Notification:warning:).*[\r\n]+([\d\D]*)(?=<!-- probot:Notification -->)[\r\n]*[\d\D]*\/i;
+ * const issueBody = `## :warning:Notification:warning:
+ * this is an example.<!-- probot:Notification -->
+ * other characters which is not important.
+ * `;
+ * const StatusData = issueBody.match(STATUS_REGEX); //new notification
+ * console.log(StatusData[2]);
+ * console.log(StatusData[2]);
+ * ```
+ * Output:
+ * ```
+ * this is an example.
+ * ```
+ *
+ */
+
+const NOTIFICATION_REGEX = /(:warning:Notification:warning:).*[\r\n]+([\d\D]*)(?=<!-- probot:Notification -->)[\r\n]*[\d\D]*/i;
+
+export enum IssueNotificationHead {
+  STATUS = `
+
+
+## Status
+
+xxx<!-- probot:Status -->
+
+`,
+
+  NOTIFICATION = `
+
+
+## :warning:Notification:warning:
+
+xxx<!-- probot:Notification -->
+
+`,
+}
+
+/**
+ * Returns a new string which updated notification in issue body.
+ * @param message - the new notification
+ * @param issueBody - issue text
+ * @param program - The chanllenge program name
+ * @return new issue body
+ */
+export function updateNotification(
```

Comment:
I don't think we need export it. And maybe the name should be createOrUpdateNotification.

Reply:
or `combineIssueContentWithNotification` ? In fact, the job of this function is generating a new issue body, the content of the issue has not actually been changed.